docs/encyclopedia/application-message-passing.mdx

```diff
@@ -220,22 +220,39 @@ Every time the Workflow wakes up--generally, it wakes up when it needs to--it wi
 
 This execution is on a single thread–while this means you don’t have to worry about parallelism, you do need to worry about concurrency if you have written Signal and Update handlers that can block. These can run interleaved with the main Workflow and with one another, resulting in potential race conditions. These methods should be made reentrant.
 
+#### Initializing the Workflow first {#workflow-initializers}
+
+Initialize your Workflow's state before handling messages. This prevents reading uninitialized instance variables.
+
+To see why, refer to the [diagram](#message-handler-concurrency). It shows that your Workflow processes messages before the first run of your Workflow's main method.
+
+This happens in two main scenarios:
+
+- When using [Signal-with-Start](#signal-with-start), the message handler always runs first.
+- When your Worker experiences delays, such as during backlogs.
+
+For object-oriented languages, use your constructor to set up state. Annotate your constructor as an Workflow Initializer and take the same arguments as your Workflow's main method.
```

Comment:
I think "object-oriented languages" on its own is not a sufficiently clear way to refer to {.NET, Java, Python}. Since it's just those 3, I vote we name them. Also, whether or not a language is "object-oriented" is not clear and sometimes argued about; Python for example need not be written in an object-oriented style. So I'd prefer a construction that gets across what we're really saying:

> languages for which Temporal uses a class to represent a Workflow together with its message handlers.

Reply:
Also PHP.  And, likely Ruby in the near future.